src/components/implementation/no_interface/scheddev/sl_lock.c

```diff
@@ -0,0 +1,110 @@
+#include "sl_lock.h"
+
+#include "sl.h"
+
+void sl_lock_init(struct sl_lock *lock) {
+    assert(lock);
+
+    *lock = (struct sl_lock) {
+        .holder = 0
+    };
+}
+
+thdid_t sl_lock_holder(struct sl_lock *lock) {
+    assert(lock);
+    return lock->holder;
+}
+
+// If timeout is 0, ignore the timeout
+int sl_lock_timeout_elapsed(microsec_t start_time, microsec_t timeout) {
+    if (timeout == 0 || (sl_now_usec() - start_time) <= timeout) {
+        return 0;
+    }
+    return 1;
+}
+
+int sl_lock_lock_internal_no_cs(struct sl_lock *lock, microsec_t timeout) {
```

Comment:
The return value indicates if the lock was successfully taken or not. Which is important if you're using timed_lock